### PR TITLE
Fix broken --exact parameter to gem command

### PR DESCRIPTION
**Pull request by Fryguy**
# Description:

Before this patch `gem list --exact` fails with the following error:

    > gem list --exact aws-sdk

    *** LOCAL GEMS ***

    ERROR:  While executing gem ... (TypeError)
        type mismatch: String given

# Tasks:

- [x] Describe the problem / feature
- [x] Write tests
- [x] Write code to solve the problem
- [x] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: lib/rubygems/commands/query_command.rb [hunk 1]
```diff
@@ -86,7 +86,7 @@ def execute
       name = Array(options[:name])
     else
       args = options[:args].to_a
-      name = options[:exact] ? args : args.map{|arg| /#{arg}/i }
+      name = options[:exact] ? args.map{|arg| /\A#{arg}\Z/i } : args.map{|arg| /#{arg}/i }
```
Comment (segiddins) [inline lib/rubygems/commands/query_command.rb]: Why does `exact` do a case insensitive match?

Comment (Fryguy) [inline lib/rubygems/commands/query_command.rb]: That's a good point...I'll change that.

**Comment by Fryguy** [comment]
@segiddins Updated.  Thanks for the review!

File: lib/rubygems/commands/query_command.rb [hunk 2]
```diff
@@ -86,7 +86,7 @@ def execute
       name = Array(options[:name])
     else
       args = options[:args].to_a
-      name = options[:exact] ? args : args.map{|arg| /#{arg}/i }
+      name = options[:exact] ? args.map{|arg| /\A#{arg}\Z/ } : args.map{|arg| /#{arg}/i }
```
Comment (segiddins) [inline lib/rubygems/commands/query_command.rb]: One last question, I'm sorry: should we `Regexp.escape` when passed exact?

Comment (Fryguy) [inline lib/rubygems/commands/query_command.rb]: Yeah, that makes sense to me.  This way it's treated like a pure string, whereas the ternary else condition there allows a user to pass an actual regex.

**Comment by Fryguy** [comment]
@segiddins Updated with the Regexp.escape.

**Comment by segiddins** [comment]
👍

**Pull request merged**